### ISSUE TITLE
Governance: store the teams in Nixpkgs

### ISSUE THREAD

**Issue by FRidh** [NixOS/nixos-homepage#645]
**Introduce your stance**
As https://github.com/NixOS/rfcs/pull/74 was after, we could use a single location where we store core project information. Governance teams and committees are part of that core information.

**Describe the issue**
Generate the teams based on the teams in the teams list of Nixpkgs.
https://github.com/NixOS/nixpkgs/blob/master/maintainers/team-list.nix

**Page links**
https://nixos.org/governance.html

**Additional context**
We need to be careful people don't add themselves to the security team.

**Comment by garbas** [comment]
@FRidh I'm not 100% convinced that a team in a `team-list.nix` should become a team on https://nixos.org/governance.html. For me `team-list.nix` shows how nixpkgs is organized and not how Nix/NixOS community is organized.

The only place I could see this useful is maybe for a specific set of teams (eg. security), but even that team might be not the same.

For lack of a better word we only use a term Team on Governance page, but I think we need to start being a bit more clear what that means and what is required for a team. I would follow a similar way that Rust community does it. A team is a more formal group of people with clear way of working and a clear set of responsibilities. And for less formal structures there are working groups.

Currently I'm a little to occupied with work to open an RFC for this, but hopefully soon or even better if somebody bits me to it.

But to move further until this organizational stuff are settled, I would create a Team called "Nixpkgs team" where we would write how `nixpkgs` is governed and organized in sub-teams.

What do you think?

**Comment by FRidh** [comment]
The reason I bring this up is that I think it should be more easily doable for people to join different teams and groups (Governance page ones, and Nixpkgs ones), which I think can be quite difficult depending on how that team or group communicates. One could of course open a new thread on Discourse and say "I would like to join..." or open a chat on IRC. 

Somehow I think that if people could see the teams in Nixpkgs, and open a PR to start discussion about joining that team, or creating a new team, lowers the threshold. This at least happened with the issue regarding commit rights.

Its important that we get more different people in these groups. They currently consist of mostly the same people that have been around for years. This *could* be a small step to get there.

> For me team-list.nix shows how nixpkgs is organized and not how Nix/NixOS community is organized.

Which is a bit of a problem. The Nixpkgs community has many more people involved in various parts, whereas the NixOS organization has not. Nixpkgs also does not have a website (anymore) to highlight these parts and the people involved. And the other parts of the Nix/NixOS community are actually relatively small, despite Nix being clearly an important core.

NixOS is in the end Nixpkgs. Hence, those groups in Nixpkgs need to be visible.

**Comment by garbas** [comment]
> The reason I bring this up is that I think it should be more easily doable for people to join different teams and groups (Governance page ones, and Nixpkgs ones), which I think can be quite difficult depending on how that team or group communicates. One could of course open a new thread on Discourse and say "I would like to join..." or open a chat on IRC.
> 
> Somehow I think that if people could see the teams in Nixpkgs, and open a PR to start discussion about joining that team, or creating a new team, lowers the threshold. This at least happened with the issue regarding commit rights.
> 
> Its important that we get more different people in these groups. They currently consist of mostly the same people that have been around for years. This _could_ be a small step to get there.
> 

I agree we need to increase the visibility and document how one can join a group. Also what are the responsibilities of that team. Right now everything is very loose and some accountability would be nice to have.

> > For me team-list.nix shows how nixpkgs is organized and not how Nix/NixOS community is organized.
> 
> Which is a bit of a problem. The Nixpkgs community has many more people involved in various parts, whereas the NixOS organization has not. Nixpkgs also does not have a website (anymore) to highlight these parts and the people involved. And the other parts of the Nix/NixOS community are actually relatively small, despite Nix being clearly an important core.
> 
> NixOS is in the end Nixpkgs. Hence, those groups in Nixpkgs need to be visible.

Well I think we are talking about a very similar stuff. 

I would really appreciate if somebody would write out how Nixpkgs is organized and how to get engaged in different teams inside nixpkgs. Also the README of nixpkgs should be organized to answer this questions as well.
